Lib/tempfile.py

```diff
@@ -130,6 +130,11 @@ def _sanitize_params(prefix, suffix, dir):
             dir = gettempdir()
         else:
             dir = gettempdirb()
+    if isinstance(dir, _os.PathLike):
```

Comment:
Why is the encoding call suddenly necessary when it wasn't needed for `dir` previously?